pkg/integration/transaction_test.go

```diff
@@ -0,0 +1,234 @@
+	require.NoError(t, err)
+	err = tx.TxSQLExec(context.TODO(), `CREATE TABLE table1(
+		id INTEGER,
+		title VARCHAR,
+		active BOOLEAN,
+		payload BLOB,
+		PRIMARY KEY id
+		);`, nil)
+	require.NoError(t, err)
+
+	params := make(map[string]interface{})
+	params["id"] = 1
+	params["title"] = "title1"
+	params["active"] = true
+	params["payload"] = []byte{1, 2, 3}
+
+	err = tx.TxSQLExec(context.TODO(), "INSERT INTO table1(id, title, active, payload) VALUES (@id, @title, @active, @payload), (2, 'title2', false, NULL), (3, NULL, NULL, x'AED0393F')", params)
+	require.NoError(t, err)
+
+	res, err := tx.TxSQLQuery(context.TODO(), "SELECT t.id as id, title FROM table1 t WHERE id <= 3 AND active = @active", params, true)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+
+	txH, err := tx.Commit(context.TODO())
+	require.NoError(t, err)
+	require.NotNil(t, txH)
+
+	err = client.CloseSession(context.TODO())
+	require.NoError(t, err)
+}
+
+func TestTransaction_Rollback(t *testing.T) {
+	options := server.DefaultOptions()
+	bs := servertest.NewBufconnServer(options)
+
+	defer os.RemoveAll(options.Dir)
+	defer os.Remove(".state-")
+
+	bs.Start()
+	defer bs.Stop()
+
+	client := ic.DefaultClient().WithOptions(ic.DefaultOptions().WithDialOptions([]grpc.DialOption{grpc.WithContextDialer(bs.Dialer), grpc.WithInsecure()}))
+
+	_, _, err := client.OpenSession(context.TODO(), []byte(`immudb`), []byte(`immudb`), "defaultdb")
+
+	tx, err := client.BeginTx(context.TODO(), &ic.TxOptions{TxMode: schema.TxMode_READ_WRITE})
+	require.NoError(t, err)
+	err = tx.TxSQLExec(context.TODO(), `CREATE TABLE table1(
+		id INTEGER,
+		PRIMARY KEY id
+		);`, nil)
+	require.NoError(t, err)
+
+	err = tx.Rollback(context.TODO())
+	require.NoError(t, err)
+
+	tx1, err := client.BeginTx(context.TODO(), &ic.TxOptions{TxMode: schema.TxMode_READ_WRITE})
+	require.NoError(t, err)
+
+	res, err := tx1.TxSQLQuery(context.TODO(), "SELECT * FROM table1", nil, true)
+	require.Error(t, err)
+	require.Equal(t, "table does not exist", err.Error())
+	require.Nil(t, res)
+
+	err = client.CloseSession(context.TODO())
+	require.NoError(t, err)
+}
+
+func TestTransaction_MultipleReadWriteError(t *testing.T) {
+	options := server.DefaultOptions()
+	bs := servertest.NewBufconnServer(options)
+
+	defer os.RemoveAll(options.Dir)
+	defer os.Remove(".state-")
+
+	bs.Start()
+	defer bs.Stop()
+
+	client := ic.DefaultClient().WithOptions(ic.DefaultOptions().WithDialOptions([]grpc.DialOption{grpc.WithContextDialer(bs.Dialer), grpc.WithInsecure()}))
+
+	serverUUID, sessionID, err := client.OpenSession(context.TODO(), []byte(`immudb`), []byte(`immudb`), "defaultdb")
+	require.NoError(t, err)
+	require.NotNil(t, serverUUID)
+	require.NotNil(t, sessionID)
+	tx1, err := client.BeginTx(context.TODO(), &ic.TxOptions{TxMode: schema.TxMode_READ_WRITE})
+	require.NoError(t, err)
+	tx2, err := client.BeginTx(context.TODO(), &ic.TxOptions{TxMode: schema.TxMode_READ_WRITE})
```

Comment:
BeginTX could produces multiple txs, and ops on tx object are good because it's possible to expose only the ones needed. You are suggesting to rename BeginTX?